paper.md

```diff
@@ -120,15 +120,91 @@ Several hosted CI services like Travis, Circle and Jenkins make continuous integ
 
 ## Workflow Definition Language
 
-## Workflow Execution Engine
-
-### **Command line interface (CLI)**
-
-### **Workflow Runner**
+YAML [@ben2009yaml] is a human-readable data-serialization language. 
+It is commonly used in writing configuration files and in applications where data is stored or transmitted. 
+Due to its simplicity and wide adoption [@yaml_wide_adoption], we chose YAML for defining popper workflows and for specifying configuration for the execution engine. 
+An example popper workflow is shown below.
+
+```yml
+steps:
+- id: download data
+  uses: docker://byrnedo/curl
+  args: [
+    "--create-dirs",
+    "-Lo data/global.csv",
+    "https://github.com/datasets/co2-fossil-global/raw/master/global.csv"
+  ]
+
+- id: run analysis
+  uses: docker://python:alpine
+  args: [
+    "scripts/get_mean_by_group.py",
+    "data/global.csv", "5"
+  ]
+
+- id: validate results
+  uses: docker://python:alpine
+  args: [
+    "scripts/validate_output.py",
+    "data/global_per_capita_mean.csv"
+  ]
+```
+
+A popper workflow consists of a series of syntactical components called steps, where each step represents a node in the workflow DAG, with a `uses` attribute specifying the required container image. 
+The `uses` attribute can reference Docker images hosted in container image registries; filesystem paths for locally defined container images (Dockerfiles); or publicly accessible GitHub repositories that contain Dockerfiles. 
+The commands or scripts that need to be executed in a container can be defined by the `args` and `runs` attributes. 
+Secrets and environment variables needed by a step can be specified by the `secrets` and `env` attributes respectively for making them available inside the container associated to a step.
+The steps in a workflow are executed sequentially in the order in which they are defined.
 
-### **Resource manager API**
+## Workflow Execution Engine
 
-### **Container engine API and plugins**
+The Popper workflow execution engine is composed of several components which talk to each other during a workflow execution.
+The vital architectural components of the system are described in detail throughout this section.
+The architecture of the Popper workflow engine is shown in @Fig:arch;
+
+### Command Line Interface (CLI)
+
+Besides allowing users to communicate with the workflow runner, the CLI provides with search capabilities so that users can search for existing actions implemented by other researchers and developers;
```

Comment:
we need to remove "users can search for existing..." as that is not possible currently